lib/galaxy/datatypes/text.py

```diff
@@ -48,12 +48,12 @@ def _looks_like_json( self, filename ):
         else:
             with open(filename, "r") as fh:
                 while True:
-                    line = fh.readline()
-                    line = line.strip()
-                    if line:
+                    # Grab first chunk of file and see if it looks like json.
+                    start = fh.read(100).strip()
+                    if start:
                         # simple types are valid JSON as well - but would such a file
                         # be interesting as JSON in Galaxy?
-                        return line.startswith("[") or line.startswith("{")
+                        return start.startswith("[") or start.startswith("{")
```

Comment:
I'm not -1, but this always seemed like a "weak" check. Maybe we should check the first two characters? There are only a couple of possibilities:

- start[0] in ('{', '[')
- start[1] in ('{', '[', '"')

which would make me feel a modicum more safe?